inspector/src/main/java/org/apache/servicecomb/inspector/internal/InspectorImpl.java

```diff
@@ -0,0 +1,201 @@
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import javax.servlet.http.Part;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response.Status;
+
+import org.apache.servicecomb.common.rest.resource.ClassPathStaticResourceHandler;
+import org.apache.servicecomb.common.rest.resource.StaticResourceHandler;
+import org.apache.servicecomb.foundation.common.part.InputStreamPart;
+import org.apache.servicecomb.inspector.internal.swagger.AppendStyleProcessor;
+import org.apache.servicecomb.inspector.internal.swagger.SchemaFormat;
+import org.apache.servicecomb.serviceregistry.RegistryUtils;
+import org.apache.servicecomb.swagger.SwaggerUtils;
+import org.apache.servicecomb.swagger.invocation.Response;
+import org.apache.servicecomb.swagger.invocation.exception.InvocationException;
+import org.asciidoctor.Asciidoctor;
+import org.asciidoctor.Asciidoctor.Factory;
+import org.asciidoctor.Attributes;
+import org.asciidoctor.AttributesBuilder;
+import org.asciidoctor.OptionsBuilder;
+import org.asciidoctor.SafeMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Ordering;
+
+import io.github.swagger2markup.Swagger2MarkupConfig;
+import io.github.swagger2markup.Swagger2MarkupConverter;
+import io.github.swagger2markup.Swagger2MarkupConverter.Builder;
+import io.github.swagger2markup.builder.Swagger2MarkupConfigBuilder;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.models.parameters.Parameter;
+
+@Path("/inspector")
+public class InspectorImpl {
+  private static final Logger LOGGER = LoggerFactory.getLogger(InspectorImpl.class);
+
+  private InspectorConfig inspectorConfig;
+
+  private Map<String, String> schemas;
+
+  private volatile Asciidoctor asciidoctor;
+
+  private StaticResourceHandler resourceHandler = new ClassPathStaticResourceHandler();
+
+  public InspectorImpl(InspectorConfig inspectorConfig, Map<String, String> schemas) {
+    this.inspectorConfig = inspectorConfig;
+    this.schemas = schemas;
+  }
+
+  @Path("/schemas")
+  @GET
+  public Collection<String> getSchemaIds() {
+    return schemas.keySet();
+  }
+
+  @Path("/download/schemas")
+  @GET
+  @ApiResponse(code = 200, message = "", response = File.class)
+  public Response downloadSchemas(@QueryParam("format") SchemaFormat format) throws IOException {
+    if (format == null) {
+      format = SchemaFormat.SWAGGER;
+    }
+
+    // normally, schema will not be too big, just save them in memory temporarily
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    ZipOutputStream zos = new ZipOutputStream(os);
```

Comment:
It's better to use try with resouce clause to make resource properly closed. And not throw IOException in REST definition.

Reply:
done
add two new commit